core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -765,7 +768,10 @@ public boolean doCheckExistingUser(String userName) throws UserStoreException {
         try {
             String searchBase = null;
             String userDN = null;
-            LdapName ldn = (LdapName)userCache.get(userName);
+            LdapName ldn = null;
+            if (userName != null) {
+                ldn = (LdapName) userCache.get(userName);
+            }
```

Comment:
It may be good to validate username once and then perform further checking instead of validating it again and again, WDYT ?